src/context/ast/QueryAstContext.h

```diff
@@ -115,14 +118,18 @@ struct MatchClauseContext final : CypherClauseContextBase {
 
     std::vector<NodeInfo>                       nodeInfos;
     std::vector<EdgeInfo>                       edgeInfos;
-    std::unordered_map<std::string, AliasType>  aliases;
     std::unique_ptr<PathBuildExpression>        pathBuild;
     std::unique_ptr<WhereClauseContext>         where;
+    std::unordered_map<std::string, AliasType>* aliasesUsed{nullptr};
```

Comment:
Could we pull the field up to the base context ?

Reply:
Currently, OrderByClauseContext and PaginationContext couldn't refer to variables produced by the preceding clauses.
I would like to handle this situation in the PR of 'unwind match'